f/briefing/f_loadoutNotes.sqf

```diff
@@ -4,144 +4,244 @@
+	};
+	if (count _mags > 0 || count _items > 0 || count _items_assigned > 0) then {
+		_textLong = _textLong + "<br/><font size='18'>OTHER [#]:</font>";
 
 		{
-			if (!(_x isEqualType []) && {_x != ""}) then {
-				_text = _text + format["<br/> |- %1",getText (configFile >> "CfgWeapons" >> _x >> "displayname")];
-			};
-		} forEach _attachments;
-
-	} forEach _weps;
-	_text = _text + "<br/>";
-};
+			_x params ["_className","_num"];
+			_textLong = _textLong + "<br/>" + ([_className, "CfgMagazines"] call _fnc_name) + format[" [%1]", _num];
+			_textShrt = _textShrt + ([_className, "CfgMagazines"] call _fnc_nameShort) + format[" [%1]", _num];
+		} forEach _mags;
 
-// ====================================================================================
-
-// OTHER MAGAZINES
-// Add lines for all magazines not tied to any carried weapon (grenades etc.)
-
-if (count _mags > 0) then {
-	_text = _text + "<br/><font size='18'>OTHER [#]:</font><br/>";
-
-	{
-		_text = _text + format["%1 [%2]<br/>",getText (configFile >> "CfgMagazines" >> _x select 0 >> "displayname"),_x select 1];
-	} forEach _mags;
-};
+		{
+			_x params ["_className","_num"];
+			_textLong = _textLong + "<br/>" + ([_className, "CfgWeapons"] call _fnc_name) + format[" [%1]",_num];
+			_textShrt = _textShrt + ([_className, "CfgWeapons"] call _fnc_nameShort) + format[" [%1]", _num];
+		} forEach _items;
+
+		if (count _mags > 0 || count _items > 0) then {
+			_textLong = _textLong + "<br/>";
+			_textShrt = _textShrt + "<br/>";
+		};
 
-// ====================================================================================
+		{
+			_textLong = _textLong + "<br/>" + ([_x, "CfgWeapons"] call _fnc_name);
+		} forEach _items_assigned;
+		
+	};
 
-// BACKPACK
-// Add lines for all other items
 
-if !(backpack player == "") then {
-	_text = _text + "<br/><font size='18'>BACKPACK [%FULL]:</font><br/>";
+	//return value
+	if (_full) exitWith {_textLong};
 
-	_bp = backpack player;
-	_text = _text + format["%1 [%2",getText (configFile >> "CfgVehicles" >> _bp >> "displayname"), 100*loadBackpack player]+"%]<br/>";
-	//_maxload = getNumber(configFile >> "CfgVehicles" >> _bp >> "maximumload");
-	//_text = _text + format["%1 [%2/%3]<br/>",getText (configFile >> "CfgVehicles" >> _bp >> "displayname"), _maxload*loadBackpack player,_maxload];
+	_textShrt
 };
 
-// ====================================================================================
+private _fnc_getLoadout = {
+	params [
+		["_unit", player, [objNull]],
+		["_full", true, [true]]
+	];
+
+	// Get a nested array containing all attached weapon items
+	private _wepItems = weaponsItems _unit;
+
+	// Get a nested array containing all unique magazines and their count
+	private _mags = (magazines _unit) call BIS_fnc_consolidateArray;
+
+	// Get a nested array containing all non-equipped items and their count
+	private _items = (items _unit) call BIS_fnc_consolidateArray;
+	private _items_assigned = assignedItems _unit;
```

Comment:
This appears to (undocumented) return Nothing when used on virtual units (Zeus and spectator). So `_items_assigned` is then (treated as) undefined in uses on L222 and L141.

Rather than address this particular issue (since it's more :enginelimitations: than anything else), I think we should instead add a guard to ensure the unit is a Man before running the loadout. Same with ORBAT, neither really make sense in the context of virtual units. See suggestions.

Reply:
sounds reasonable? Although the ORBAT can also be useful (sometimes) for the zeus?

Reply:
It's empty though (only does same side, so sideLogic)

Reply:
oh right. then it shouldn't be there at all.

Reply:
we can revisit zeus briefing at a later point (possibly showing ORBAT/Gear of players?)

Reply:
Yeah, sounds like an idea to explore for a later update.